### PR TITLE
Add clang-format tool

### PR DESCRIPTION
**Pull request by steveire**

### CODE REVIEW

File: lib/tooling/clang-format-tool.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2018, Compiler Explorer Authors
```
Comment (RabsRincon) [inline lib/tooling/clang-format-tool.js]: 2019

**Comment by steveire** [comment]
> It's currently missing the config changes to add it to each language though.

I'm not certain how to do this for the CE production setup. Can you help?

**Comment by mattgodbolt** [comment]
> I'm not certain how to do this for the CE production setup. Can you help?

In honesty I'm not 100% sure, but for example in the `c++.amazon.properties` file there are lines like:

```
tools=clangtidytrunk:llvm-mcatrunk:pahole:clangquerytrunk:readelf:x86to6502:ldd:iwyu

tools.clangtidytrunk.exe=/opt/compiler-explorer/clang-trunk/bin/clang-tidy
tools.clangtidytrunk.name=clang-tidy (trunk)
tools.clangtidytrunk.type=independent
tools.clangtidytrunk.class=clang-tidy-tool
tools.clangtidytrunk.exclude=cl19:cl_new
tools.clangtidytrunk.options=--gcc-toolchain=/opt/compiler-explorer/gcc-8.2.0
tools.clangtidytrunk.stdinHint=disabled
```

...and I suspect we'll need to add clang-format here too.

File: lib/tooling/clang-format-tool.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        const dir = path.dirname(sourcefile);
+
+        const compileFlags = options.filter(option => (option !== sourcefile));
+        if (!compilerExe.includes("clang++")) {
+            compileFlags.push(this.tool.options);
+        }
+
+        await fs.writeFile(path.join(dir, "compile_flags.txt"), compileFlags.join('\n'));
+        args.push('-style={' + stdin.trim().replace("\n", ",") + '}');
+        const toolResult = await super.runTool(compilationInfo, sourcefile, args, stdin);
```
Comment (mattgodbolt) [inline lib/tooling/clang-format-tool.js]: Total nitpick here, but one can return `await super.runTool` here. Saves a line :P

**Comment by mattgodbolt** [comment]
Any way I can help get this moving? I don't want it to stall on our account :)

**Comment by steveire** [comment]
Should be ready now, thanks,

**Comment by RabsRincon** [comment]
Looks good, thank you! It should hit the live site shortly :)

**Pull request merged**